contracts/CrossChain.sol

```diff
@@ -322,9 +409,143 @@ contract CrossChain is System, ICrossChain, IParamSubscriber{
         registeredContractChannelMap[handlerContract][channelId] = isEnable;
         emit enableOrDisableChannel(channelId, isEnable);
       }
+    } else if (Memory.compareStrings(key, "suspendQuorum")) {
+      require(value.length == 2, "length of value for suspendQuorum should be 2");
+      uint16 suspendQuorum = BytesToTypes.bytesToUint16(32, value);
+      require(suspendQuorum > 0, "zero suspend quorum");
+      quorumMap[SUSPEND_PROPOSAL] = suspendQuorum;
+    } else if (Memory.compareStrings(key, "reopenQuorum")) {
+      require(value.length == 2, "length of value for reopenQuorum should be 2");
+      uint16 reopenQuorum = BytesToTypes.bytesToUint16(32, value);
+      require(reopenQuorum > 0, "zero reopen quorum");
+      quorumMap[REOPEN_PROPOSAL] = reopenQuorum;
+    } else if (Memory.compareStrings(key, "cancelTransferQuorum")) {
+      require(value.length == 2, "length of value for cancelTransferQuorum should be 2");
+      uint16 cancelTransferQuorum = BytesToTypes.bytesToUint16(32, value);
+      require(cancelTransferQuorum > 0, "zero cancel transfer quorum");
+      quorumMap[CANCEL_TRANSFER_PROPOSAL] = cancelTransferQuorum;
     } else {
       require(false, "unknown param");
     }
     emit paramChange(key, value);
   }
-}
\ No newline at end of file
+
+  // BEP-171: Security Enhancement for Cross-Chain Module
+  function challenge(
+    // to avoid stack too deep error, using `uint64[4] calldata params`
+    // instead of  `uint64 height0, uint64 height1, uint64 packageSequence, uint8 channelId`
+    uint64[4] calldata params, // 0-height0, 1-height1, 2-packageSequence, 3-channelId,
```

Comment:
I think we use the same height is enough

Reply:
If we enforce use the same height,  we must run a BC archive node, we don't have one yet.